src/snowflake/snowpark/_internal/utils.py

```diff
@@ -1924,6 +1926,144 @@ def remove_comments(sql_query: str, uuids: List[str]) -> str:
     )
 
 
+class TimeTravelConfig(NamedTuple):
+    """Configuration for time travel operations."""
+
+    mode: str  # 'at' or 'before'
+    statement: Optional[str] = None
+    offset: Optional[int] = None
+    timestamp: Optional[str] = None
+    timezone: str = "NTZ"
+    stream: Optional[str] = None
+
+
+def validate_and_normalize_time_travel_params(
```

Comment:
have you though about making this a static method of `TimeTravelConfig` class? seems like they are closely related.